src/index.d.ts

```diff
@@ -0,0 +1,13 @@
+export as namespace parseDomain;
+
+declare namespace parseDomain {
+  interface IParsedDomain {
+    domain?: string;
+    subdomain?: string;
+    tld?: string;
+  }
+}
+
+declare function parseDomain (url: string, customTlds?: RegExp | string[], privateTlds?: boolean): parseDomain.IParsedDomain;
```

Comment:
`customTlds` and `privateTlds` should be part of an `ParseOptions` object. Take a look at the README :)